Sources/PayPalNativePayments/PayPalNativeCheckoutClient.swift

```diff
@@ -42,7 +42,7 @@ public class PayPalNativeCheckoutClient {
     public func start(
         request: PayPalNativeCheckoutRequest,
         presentingViewController: UIViewController? = nil
-    ) async {
+    ) {
```

Comment:
We will want to add a CHANGELOG entry for this since it's a public method

Reply:
That's a good call. Is this considered a breaking change in iOS?

Reply:
Yes this is a breaking change, if we made this change a merchant app would not compile as-is

Reply:
I would deprecate the old method until we're ready for V2

Reply:
Definitely yeah that makes sense 👍 .

Reply:
Steven and I tried creating another function with same name with async and annotate it as deprecated.
There is a problem with the calling function not knowing whether it should be async or not. 
So if we do this, we need to rename the new non async method.

Reply:
I think the goal is to _not_ call this in the task since it shouldn't be async. We can either restructure how we are calling things in the demo to separate out the order ID call or add a `DispatchSemaphore` to wait for the async to `getOrderID` to finish:
```swift
    private func checkout(_ shippingPreference: ShippingPreference) {
        state = .loading(content: "Initializing checkout")
        let semaphore = DispatchSemaphore(value: 0)

        Task {
            do {
                orderID = try await self.getOrderID(shippingPreference)
                self.shippingPreference = shippingPreference
                semaphore.signal()
            } catch let error {
                publishStateToMainThread(.mainContent(title: "Error", content: "\(error.localizedDescription)", flowComplete: true))
            }
        }

        semaphore.wait()
        let request = PayPalNativeCheckoutRequest(orderID: orderID)
        payPalClient?.start(request: request)
    }
```

Reply:
Hey Jax, do we really want to use semaphore? This function is reliant on result from getOrderID, that's why it's in the same Task block.

Reply:
I am able to replicate the weirdness you're seeing Victoria. It looks like the compiler _is_ able to distinguish between `func start() async` and `func start()` only if you move the non-async one out of a Task like in Jax's example. 

I also wasn't expecting that behavior, but it seems to be by design [(see Swift proposal where this got added)](https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md#overloading-and-overload-resolution)

Reply:
Yes, it's a weird assumption on Swift compiler since Task block is supposed to handle both async and sync functions.

Reply:
We don't have to as I mentioned in my comment you can also restructure the method calls. Though a semaphore is calling down to the kernel only when the calling thread needs to be blocked, which as you mentioned is the goal since we need the result from the `getOrderID` call to call `start`. 

Is there a specific reason why the `getOrderID` call is happening at the start of the checkout flow instead of earlier? Looking though PayPal Web for example, we have this as a separate step/view? If anything I think `getOrderID` happening in this same task is an architecture issue since this is our only view/viewmodel that appears to do it at this point.

Reply:
I don't think there was a good reason for this in the demo app except maybe there is extra step of verifying clientID sent into for native checkout and maybe it was decided not to have that extra step to check orderID?

I can make changes in another ticket for it.

Reply:
I guess that is a good solution for not having to rename the function.

Reply:
I don't think it's confusing to use the same method name for the async/non async method. This also means we won't have to go back and change the name to the one we actually want once we remove the deprecated method.

Reply:
It is pretty annoying that Xcode won't let you use the non-async `start()` function in a Task just because one exists with a duplicate name. That isn't the behavior of a method without a duplicate name.

Also this is the first time we're dealing with the fact that Swift doesn't support silencing deprecation warnings via Macros ([example](https://stackoverflow.com/questions/31540446/how-to-silence-a-warning-in-swift)).

We _could_ leave a `NEXT_MAJOR_VERSION` note to change this function signature later on when we're ready for v2?

Reply:
Steven's vote was for change in v2 since it is a breaking change.

Reply:
Cool, I'm down if we want to leave a `NEXT_MAJOR_VERSION`. We use that tag so that when time comes for doing a breaking major version release we have all of our TODOs related to it easily searchable

Reply:
Sorry, but is it just comment // NEXT_MAJOR_VERSION? I don't think I've seen it.

Reply:
Just noting that we can have the two separate async and non-async functions if we rename the new function as I did.